radio/src/audio_arm.cpp

```diff
@@ -511,9 +515,16 @@ void audioTask(void * pdata)
 }
 #endif
 
-void mixSample(audio_data_t * result, int sample, unsigned int fade)
+void mixSample(audio_data_t * result, int sample, unsigned int fade, bool swVolumeAdjust = true)
 {
-  *result = limit(AUDIO_DATA_MIN, *result + ((sample >> fade) >> (16-AUDIO_BITS_PER_SAMPLE)), AUDIO_DATA_MAX);
+  *result = limit(AUDIO_DATA_MIN, *result + ((sample >> fade) >> (16 - AUDIO_BITS_PER_SAMPLE)), AUDIO_DATA_MAX);
+
+#if defined(SOFTWARE_VOLUME)
+  if (!swVolumeAdjust)
+    return;
+  int32_t tmpSample = (int32_t) (uint32_t) (*result);  // conversion from uint16_t
+  *((uint16_t*) result) = (int16_t) ((tmpSample * currentSpeakerVolume) / VOLUME_LEVEL_MAX);
```

Comment:
The problem is here. The zero level audio signal (silence) has a value of AUDIO_DATA_SILENCE, so to convert from unsigned sample value you have to subtract the AUDIO_DATA_SILENCE. Then you do the scaling, and then convert back to unsigned value (add back AUDIO_DATA_SILENCE).

Example of the first part is in simpgmspace.cpp:
```c
    int sample = ((int32_t)(uint32_t)(buff[i]) - 0x8000);  // conversion from uint16_t
```

And like Bertrand suggested, do the sw volume at the end, after all the mixing was done.